cmds/fmap/lib/fmap.go

```diff
@@ -22,15 +22,19 @@ const (
 	FmapAreaReadOnly
 )
 
-type FMap struct {
-	Start     uint64 // not part of the written struct, but useful nevertheless
+type fMapInternal struct {
 	Signature [8]uint8
 	VerMajor  uint8
 	VerMinor  uint8
 	Base      uint64
 	Size      uint32
 	Name      [32]uint8
-	Areas     []FMapArea
+}
+
+type FMap struct {
+	fMapInternal
+	Start uint64
```

Comment:
You are right that I have not completely thought through encoding the struct back to binary. While I write the `WriteFMap` function, I might refactor it a bit. I got some unit tests, so it is fairly easy to change.

`Start` is a piece of metadata (the location of the fmap) the `ReadFMap` function returns. It is fairly useful in the printed output. It might be better to return two values, an FMap and a FMapMetadata. What do you think?